components/ActionModals/ClaimModal.vue

```diff
@@ -0,0 +1,165 @@
+<template>
+  <ActionModal
+    id="modal-withdraw-rewards"
+    ref="actionModal"
+    :transaction-data="transactionData"
+    :notify-message="notifyMessage"
+    :selected-denom="feeDenom"
+    title="Claim Rewards"
+    class="modal-withdraw-rewards"
+    submission-error-prefix="Withdrawal failed"
+    feature-flag="claim_rewards"
+    :transaction-type="lunieMessageTypes.CLAIM_REWARDS"
+    :rewards="rewards"
+    :disable="validatorsWithRewards"
+  >
+    <span class="form-message withdraw-limit">
+      Lunie will only withdraw rewards from 5 validators at a time because of a
+      limitation with the Ledger Nano&nbsp;S.
+    </span>
+    <FormGroup
+      class="action-modal-form-group"
+      field-id="amount"
+      :field-label="`Rewards from ${top5Validators.length} ${
+        top5Validators.length > 1 ? `validators` : `validator`
+      }`"
+    >
+      <div
+        v-for="reward in totalRewards"
+        :key="reward.denom"
+        class="rewards-list-item"
+      >
+        <Field disabled="disabled" :value="reward.amount | fullDecimals" />
+        <span class="input-suffix">{{ reward.denom }}</span>
+      </div>
+    </FormGroup>
+  </ActionModal>
+</template>
+
+<script>
+import { fullDecimals } from '~/common/numbers'
+import { getTop5RewardsValidators } from '~/signing/transaction-manager'
+import { lunieMessageTypes } from '~/common/lunie-message-types'
+import network from '~/common/network'
+
+function rewardsToDictionary(rewards) {
+  return rewards.reduce((all, reward) => {
+    return {
+      ...all,
+      [reward.denom]: Number(reward.amount) + (all[reward.denom] || 0),
+    }
+  }, {})
+}
+
+export default {
+  name: `ClaimModal`,
+  filters: {
+    fullDecimals,
+  },
+  props: {
+    address: {
+      type: String,
+      required: true,
+    },
+    rewards: {
+      type: Array,
+      required: true,
+    },
+    balances: {
+      type: Array,
+      required: true,
+    },
+  },
+  data: () => ({
+    getTop5RewardsValidators,
+    lunieMessageTypes,
+    network,
+  }),
+  computed: {
+    transactionData() {
+      if (this.totalRewards.length === 0) return {}
+      return {
+        type: lunieMessageTypes.CLAIM_REWARDS,
+        amounts: this.totalRewards,
+        from: this.top5Validators,
+      }
+    },
+    top5Validators() {
+      if (this.rewards && this.rewards.length > 0) {
+        const top5Validators = this.getTop5RewardsValidators(this.rewards)
+        return top5Validators
+      } else {
+        return []
+      }
+    },
+    notifyMessage() {
+      return {
+        title: `Successful withdrawal!`,
+        body: `You have successfully withdrawn your rewards.`,
+      }
+    },
+    validatorsWithRewards() {
+      if (this.rewards) {
+        return this.rewards.length > 0
+      } else {
+        return false
+      }
+    },
+    feeDenom() {
```

Comment:
can we simplify this now?

Reply:
well, do you know if in Stargate you can pay fees with IBC tokens? Probably, no? Like in Terra and e-Money

Reply:
Fee denoms get selected in the table invoice as far as I know

Reply:
actually I'm thinking, this shouldn't be here. It should simply default to stakingDenom :thinking: Since this is the only currency you can stake with & logically claim

Reply:
we have that logic only for SendModal. Let me check